tests/test_std.rs

```diff
@@ -620,12 +627,17 @@ fn combinations_with_replacement() {
     // Zero size
     it::assert_equal(
         (0..3).combinations_with_replacement(0),
-        <Vec<Vec<_>>>::new(),
+        vec![vec![]],
+    );
+    // Zero size on empty pool
+    it::assert_equal(
+        (0..0).combinations_with_replacement(0),
+        vec![vec![]],
     );
     // Empty pool
     it::assert_equal(
         (0..0).combinations_with_replacement(2),
-        <Vec<Vec<_>>>::new(),
+        vec![vec![]],
```

Comment:
Now this is wrong. `vec![]` is only a valid combination of length 0; it’s not a valid combination of length 2.